### PR TITLE
Make FlutterBuildStatus autotrigger on listen

### PR DESCRIPTION
**Pull request by Hixie**
...rather than having an explicit init step

### CODE REVIEW

**Comment by Hixie** [comment]
cc @CaseyHillers same as the previous PR but for build state.

File: app_flutter/lib/state/flutter_build.dart [hunk 1]
```diff
@@ -73,69 +64,101 @@ class FlutterBuildState extends ChangeNotifier {
...
-    _fetchBuildStatusUpdate();
+  /// Timer that calls [_fetchStatusUpdates] on a set interval.
+  @visibleForTesting
+  @protected
+  Timer refreshTimer;
 
-    _fetchFlutterBranches().then((List<String> branchResponse) => _branches = branchResponse);
+  // There's no way to cancel futures in the standard library so instead we just track
+  // if we've been disposed, and if so, we drop everything on the floor.
+  bool _active = true;
```
Comment (CaseyHillers) [inline app_flutter/lib/state/flutter_build.dart]: Obviously we implement this with Streams then :P

File: app_flutter/lib/state/flutter_build.dart [hunk 2]
```diff
@@ -73,69 +64,101 @@ class FlutterBuildState extends ChangeNotifier {
...
-      cocoonService.fetchCommitStatuses(branch: _currentBranch).then((CocoonResponse<List<CommitStatus>> response) {
+  Future<void> _fetchStatusUpdates([Timer timer]) async {
+    await Future.wait<void>(<Future<void>>[
+      () async {
+        final CocoonResponse<List<String>> response = await cocoonService.fetchFlutterBranches();
+        if (!_active) {
+          return null;
+        }
+        if (response.error != null) {
+          _errors.send('$errorMessageFetchingBranches: ${response.error}');
```
Comment (CaseyHillers) [inline app_flutter/lib/state/flutter_build.dart]: the error brook handles notifying it's listeners, right? (I saw the notifyListener was brought into the else logic block)

File: app_flutter/lib/state/flutter_build.dart (hunk 1 above)
Comment (Hixie) [reply app_flutter/lib/state/flutter_build.dart]: Or some sort of cancelable API, yeah. We could. Not sure it's really worth it though. At the end of the day this boils down to a network request and it's not clear how much we win from canceling that.

File: app_flutter/lib/state/flutter_build.dart (hunk 2 above)
Comment (Hixie) [reply app_flutter/lib/state/flutter_build.dart]: Yeah, the error brook has a different set of listeners than the state.

**Pull request merged**